### PR TITLE
Controller dto mapper task

### PR DESCRIPTION
**Pull request by vadymhrnk**

### CODE REVIEW

File: src/main/java/com/example/bookstore/service/impl/BookServiceImpl.java [hunk 1]
```diff
@@ -1,27 +1,40 @@
...
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public BookDto findById(Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't get book by id: " + id)
+        );
+        return bookMapper.toDto(book);
```
Comment (TarasYurkovskyi) [inline src/main/java/com/example/bookstore/service/impl/BookServiceImpl.java:38]: your method named find but "Can't get book by id"

**Review by TarasYurkovskyi — APPROVED** [review]
Good job)

File: src/main/java/com/example/bookstore/repository/impl/BookRepositoryImpl.java [hunk 2]
```diff
@@ -32,7 +30,7 @@ public Book save(Book book) {
             if (transaction != null) {
                 transaction.rollback();
             }
-            throw new RuntimeException("Can't add a new book: " + book, e);
+            throw new EntityNotFoundException("Can't add a new book: " + book, e);
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/repository/impl/BookRepositoryImpl.java]: Does `EntityNotFoundException` suits for this case?

File: src/main/java/com/example/bookstore/repository/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -47,7 +45,17 @@ public List<Book> findAll() {
...
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Session session = factory.openSession()) {
+            Book book = session.find(Book.class, id);
+            return Optional.ofNullable(book);
+        } catch (Exception e) {
+            throw new EntityNotFoundException("Cant get book by id: " + id, e);
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/repository/impl/BookRepositoryImpl.java]: Does `EntityNotFoundException` suits for this case?

File: src/main/java/com/example/bookstore/service/impl/BookServiceImpl.java (hunk 1 above)
Comment (vovasalyha) [reply src/main/java/com/example/bookstore/service/impl/BookServiceImpl.java:38]: Makes sense. Semantically it would be more correct.

**Review by vovasalyha — CHANGES_REQUESTED** [review]
All good, but see few comments.

**Review by kshuryhin — APPROVED** [review]
Looks good

**Pull request merged**